eco-core/core-plugin/src/main/resources/enchants/smelting.yml

```diff
@@ -0,0 +1,26 @@
+display-name: "Smelting" 
+description: "Automatically smelts mined blocks." 
+placeholder: [ ]
+placeholders: [ ]
+type: normal
+
+targets: 
+  - axe
+  - shovel
+  - pickaxe
+conflicts: [ ]
+rarity: veryspecial
+max-level: 1
+
+tradeable: true 
+discoverable: true
+enchantable: true 
+
+effects:
+  - id: autosmelt
+    args:
+      drop_xp: true
+

```

Comment:
Pretty sure you need a item drop event trigger

effects:
  - id: autosmelt
    args:
      drop_xp: true
    triggers:
      - block_item_drop